.github/actions/dockerfiles/Dockerfile.blocklist-client.debian

```diff
@@ -1,32 +1,68 @@
-FROM rust:1.91.0-slim-bookworm AS build
+FROM --platform=$BUILDPLATFORM rust:1.91.0-slim-bookworm AS build
 
+# Docker automatically populates these arguments:
+ARG TARGETARCH
 ARG CARGO_BUILD_ARGS="--release --locked"
 
-# Install dependencies.
-RUN apt-get update
+# 1. Setup Debian for Multi-Architecture
+# We enable arm64 so we can install ARM system libraries (OpenSSL, glibc)
+RUN dpkg --add-architecture arm64 && \
+    dpkg --add-architecture amd64 && \
+    apt-get update
+
+# 2. Install Host Dependencies (Tools needed to RUN the build on AMD64)
 RUN apt-get install -y --no-install-recommends \
-    libclang-dev \
     git \
     pkg-config \
-    libssl-dev \
     make \
     protobuf-compiler \
     npm \
     default-jre \
-    g++ && \
-    apt-get clean && rm -rf /var/lib/apt/lists/*
-
-RUN npm install -g pnpm@9
-RUN npm install -g @openapitools/openapi-generator-cli
+    curl \
+    g++ \
+    libssl-dev \
+    clang \
+    libclang-dev \
+    findutils
+    
+# 3. Install Cross-Compilation Toolchains & Target Dependencies
+RUN apt-get install -y --no-install-recommends \
+    gcc-aarch64-linux-gnu \
+    g++-aarch64-linux-gnu \
+    libc6-dev-arm64-cross \
+    libssl-dev:arm64 \
+    && apt-get clean && rm -rf /var/lib/apt/lists/*
+    
+RUN npm install -g pnpm@9 && \
+    npm install -g @openapitools/openapi-generator-cli
 
 WORKDIR /code/sbtc
 COPY . .
-RUN make install-pnpm && make build
-RUN cargo build --bin blocklist-client ${CARGO_BUILD_ARGS}
 
-# Create Docker image to run the signer.
+RUN export LIBCLANG_PATH="$(find /usr/lib/llvm-* -name lib -type d | head -n 1)" && \
+    make install-pnpm && \
+    make blocklist-client-codegen emily-client-codegen contracts
```

Comment:
Do we need these two `make`? Looking at the blocklist client deps it's not obvious we do

Reply:
[8e8e39c](https://github.com/stacks-sbtc/sbtc/pull/1874/commits/8e8e39c5a92c6e41c8beb49fc5c472004e433e1c)